src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/NewStyleOneTimeLinkController.kt

```diff
@@ -14,27 +21,46 @@ import java.time.Duration
 
 class NewStyleOneTimeLinkController(
         context: ActionContext,
-        private val repo: TokenRepository,
+        private val tokenRepository: TokenRepository,
+        private val userRepository: UserRepository,
+        private val emailManager: EmailManager = getEmailManager(),
         private val tokenHelper: WebTokenHelper = WebTokenHelper(KeyHelper.keyPair),
         private val serializer: Serializer = MontaguSerializer.instance,
         private val redirectValidator: RedirectValidator = RedirectValidator())
     : Controller(context)
 {
     constructor(context: ActionContext, repositories: Repositories)
-            : this(context, repositories.token)
+            : this(context, repositories.token, repositories.user)
 
     fun getTokenForDemographicData(): String
     {
         return getOneTimeLinkToken(OneTimeAction.DEMOGRAPHY)
     }
 
+    fun createUser(): String
+    {
+        val user = context.postData<CreateUser>()
+        userRepository.addUser(user)
+
+        val params = mapOf(":username" to user.username)
+
+        val token = getOneTimeLinkToken(OneTimeAction.SET_PASSWORD,
+                username = user.username,
+                params = params,
+                duration = Duration.ofDays(1))
+
+        emailManager.sendEmail(NewUserEmail(user, token), user)
+        return objectCreation(context, "/users/${user.username}/")
+    }
```

Comment:
Rather than putting this method in `NewStyleOneTimeLinkController`, maybe just factor out `getOneTimeLinkToken` into a helper class that both this controller and `UserController` new up as needed?

Reply:
Yes, good shout. Having started refactoring the Password controller I've realised that this will also come in handy there.